src/libraries/System.Private.CoreLib/src/System/Threading/RegisteredWaitHandle.WindowsThreadPool.cs

```diff
@@ -43,6 +44,9 @@ public sealed partial class RegisteredWaitHandle : MarshalByRefObject
 
             _tpWait = Interop.Kernel32.CreateThreadpoolWait(&RegisteredWaitCallback, (IntPtr)_gcHandle, IntPtr.Zero);
 
+            if (NativeRuntimeEventSource.Log.IsEnabled())
+                NativeRuntimeEventSource.Log.ThreadPoolIOEnqueue(this);
```

Comment:
Could move this to the end of the method after the error check, it looks a bit awkward to separate the error checking for the above code with some other code